File: Java/Linked List/Remove duplicates from an unsorted linked list.java
class Solution
{
    //Function to remove duplicates from unsorted linked list.
    public Node removeDuplicates(Node head) 
    {
         // Your code here
         HashSet<Integer> hs = new HashSet<>();
     
        /* Pick elements one by one */
        Node current = head;
        Node prev = null;
        while (current != null)
        {
            int curval = current.data;
             
             // If current value is seen before
            if (hs.contains(curval)) {
                prev.next = current.next;
            } else {
                hs.add(curval);
                prev = current;
            }
            current = current.next;
        }
        return head;
    }
}

File: C++/Linked List/Remove duplicates from an unsorted linked list.cpp
class Solution
{
    public:
    //Function to remove duplicates from unsorted linked list.
    Node * removeDuplicates( Node *head) 
    {
     // your code goes here
        unordered_set<int> seen;
 
        /* Pick elements one by one */
        struct Node *curr = head;
        struct Node *prev = NULL;
        while (curr != NULL)
        {
            // If current value is seen before
            if (seen.find(curr->data) != seen.end())
            {
               prev->next = curr->next;
               delete (curr);
            }
            else
            {
               seen.insert(curr->data);
               prev = curr;
            }
            curr = prev->next;
        }
        return head;
    }
};
